libvast/vast/concept/printable/core/printer.hpp

```diff
@@ -40,30 +40,30 @@ struct printer_base {
   }
 
   // FIXME: don't ignore ADL.
-  template <class Range, class Attribute = unused_type>
-  auto operator()(Range&& r, const Attribute& a = unused) const
-  -> decltype(std::begin(r), std::end(r), bool()) {
-    auto out = std::back_inserter(r);
-    return derived().print(out, a);
+  template <class Range, class... TAttributes>
+    requires requires(Range r) {
+      std::begin(r);
+      std::end(r);
+    }
+  auto operator()(Range&& r, const TAttributes&... attributes) const -> bool {
```

Comment:
Nit: this can now be called without an attribute, I guess it would fail at `std::tie` then.
Nit: There is no reason to use a trailing return type any more.

Reply:
The same applies to the overload for iterators.

Reply:
Just pushed a change that adds a constraint the `sizeof...(TAttributes) > 0` to give a better error message if someone were to call this call operator without any attributes.